File: src/cinemol_py/cine_mol_helpers.py
from math import pow
from .fable_modules.fable_library.double import sqrt
from .fable_modules.fable_library.util import round as round_1

def round(digits: int, v: float) -> float:
    return round_1(v, digits)


def abs_1(v: float) -> float:
    return sqrt(pow(v, 2.0))


def clamp(lower_bound: float, upper_bound: float, v: float) -> float:
    if v < lower_bound:
        return lower_bound

    elif v > upper_bound:
        return upper_bound

    else: 
        return v



__all__ = ["round", "abs_1", "clamp"]

File: src/_CineMol/CineMol.Helpers.fs
module CineMol.Helpers

open System

/// <summary>
/// Round floating point number to desired number of digits.
/// </summary>
let round (digits: int) (v: float) = Math.Round(v, digits)

/// <summary>
/// Take absolute value of floating point number.
/// </summary>
let abs (v: float) = Math.Sqrt(v ** 2.0)

/// <summary>
/// Clamp a floating point value between two other floating point values.
/// </summary>
let clamp lowerBound upperBound (v: float) =
    if v < lowerBound then lowerBound
    elif v > upperBound then upperBound
    else v
    
/// <summary>
/// Try to cast string.
/// </summary>
let inline tryCast (cast: string -> 'a) s =
    try s |> cast |> Some 
    with :? FormatException -> None
    
/// <summary>
/// Enumerate a list.
/// </summary>
let inline enumerate (items: 'a list) = List.zip [0 .. 1 .. items.Length] items 